lib/compress.js

```diff
@@ -338,7 +338,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant(compressor)) {
+                    if (!(var_decl.value instanceof AST_RegExp) && var_decl.value.is_constant(compressor)) {
```

Comment:
It's correct as is. If you studied the code, this is the other use of is_constant:
```
echo 'function f(x) { return x ? /foo/ : /foo/; }' | uglifyjs -c
function f(x){return/foo/}
```